pkg/rhRegion/rhRegion.go

```diff
@@ -0,0 +1,50 @@
+package rhRegion
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	"os"
+	"strings"
+)
+
+type Region struct {
+	URL string
+	AWS []string
+	GCP []string
+}
+
+func GetRhRegions(ctx context.Context, token string) (map[string]Region, error) {
+	var regions map[string]Region
+	globalConn, err := sdk.NewConnectionBuilder().Tokens(token).RhRegion("integration").BuildContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer globalConn.Close()
+	resp, err := globalConn.Get().Path("/static/ocm-shards.json").SendContext(ctx)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't retrieve shards: %s\n", err)
+	}
+	err = json.Unmarshal(resp.Bytes(), &regions)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't unmarshal shards: %s\n", err)
+	}
+
+	return regions, nil
+}
+
+func GetRhRegion(ctx context.Context, token string, regionName string) (Region, error) {
+	regions, err := GetRhRegions(ctx, token)
+	if err != nil {
+		return Region{}, err
+	}
+	if !strings.HasPrefix(regionName, "rh-") {
```

Comment:
I understand the impulse to be flexible or helpful, but in general try to avoid "smart keys" where you support multiple key variations with application logic, it ends up being a pain in the butt to maintain long term and is generally unintuitive to users. 

Speaking from experience :)

Reply:
Removed the rh- prefix check